ete4/smartview/ete/gardening.pyx

```diff
@@ -52,18 +40,16 @@ def add_intermediate(node):
     "Add an intermediate parent to the given node and return it"
     parent = node.up
 
-    pos = parent.children.index(node)  # position of node in parent's children
-
-    parent.children.pop(pos)  # detach from parent
+    parent.children.remove(node)  # detach from parent
```

Comment:
We didn't really need to reposition the intermediate node at the exact same place in the children list, so this seems more straightforward.